htdocs/core/tpl/objectline_view.tpl.php

```diff
@@ -182,6 +182,22 @@
 		if ($line->date_start || $line->date_end) {
 			print '<div class="clearboth nowraponall opacitymedium">'.get_date_range($line->date_start, $line->date_end, $format).'</div>';
 		}
+		if (!$line->date_start || !$line->date_end) {
+			// show warning under line
+			// we need to fetch product associated to line for some test
+			if ($object->element != 'order_supplier') {
```

Comment:
Why test 
if ($object->element != 'order_supplier') {
and not 
supplier_proposal or and any other possible values ?
It is a generic class and lot of external module also use this tpl file;

May be we should invert the test to make the field mandatory when it is in a list of known object (proposal, sale order and invoices only) ?

Reply:
made changes